torchrl/data/tensordict/tensordict.py

```diff
@@ -1610,6 +1615,8 @@ def __setitem__(self, index: INDEX_TYPING, value: TensorDictBase) -> None:
                 raise err
         else:
             indexed_bs = _getitem_batch_size(self.batch_size, index)
```

Comment:
I've put the TensorDict casting over here, because the batch size wouldn't be computed correctly for broadcasting